app/models/finance/billing_strategy.rb

```diff
@@ -323,19 +324,23 @@ def charge_invoices(buyer, now = Time.now.utc)
       end
 
       buyer.invoices.chargeable(now).find_each(batch_size: 50) do |invoice|
-        Rails.logger.info("Trying to charge invoice #{invoice.id}")
+        Rails.logger.info("#{log_prefix(buyer)} trying to charge invoice #{invoice.id}")
         invoice.charge!
 
       end
     else
-      Rails.logger.info("Charging not enabled for #{account.org_name} - bypassing")
+      Rails.logger.info("#{log_prefix(buyer)} charging not enabled for provider #{account.org_name} - bypassing")
     end
   end
 
   def needs_credit_card?
     charging_enabled?
   end
 
+  def log_prefix(buyer)
+    "[billing] provider #{buyer.provider_account_id} buyer #{buyer.id}:"
```

Comment:
IIRC daily depending on the options may or may not have a buyer. I'm not sure, have to check later. But is it sure we have a buyer?

Reply:
`#daily` indeed not requires buyer ID, it's an option:
https://github.com/3scale/porta/blob/improve-billing-logging/app/models/finance/billing_strategy.rb#L352-L365

However the logs that use `buyer` do have buyer always available, because it's within the `yield(buyer)` block